lib/vagrant/machine.rb

```diff
@@ -10,6 +10,8 @@ module Vagrant
   # API for querying the state and making state changes to the machine, which
   # is backed by any sort of provider (VirtualBox, VMware, etc.).
   class Machine
+    extend Vagrant::Action::Builtin::MixinSyncedFolders
```

Comment:
I think I'm not a huge fan of extending from a module in `Vagrant::Action` here.
I think what I really want here is to have the bit of `Vagrant::Action::Builtin::MixinSyncedFolders.synced_folders` be defined here so that you can do a `machine.synced_folders`.